account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -0,0 +1,158 @@
+import type {
+  EntryPointDef,
+  SmartAccountSigner,
+  SmartContractAccountWithSigner,
+  ToSmartContractAccountParams,
+} from "@aa-sdk/core";
+import {
+  createBundlerClient,
+  getAccountAddress,
+  getEntryPoint,
+  toSmartContractAccount,
+} from "@aa-sdk/core";
+import {
+  concatHex,
+  encodeFunctionData,
+  getContract,
+  type Address,
+  type Chain,
+  type Hex,
+  type Transport,
+} from "viem";
+import { accountFactoryAbi } from "../abis/accountFactoryAbi.js";
+import { addresses } from "../utils.js";
+import { standardExecutor } from "../../msca/account/standardExecutor.js";
+import { singleSignerMessageSigner } from "../modules/single-signer-validation/signer.js";
+
+export const DEFAULT_OWNER_ENTITY_ID = 0;
+
+export type SMAV2Account<
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = SmartContractAccountWithSigner<"SMAV2Account", TSigner, "0.7.0">;
+
+export type CreateSMAV2AccountParams<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner,
+  TEntryPointVersion extends "0.7.0" = "0.7.0"
+> = Pick<
+  ToSmartContractAccountParams<
+    "SMAV2Account",
+    TTransport,
+    Chain,
+    TEntryPointVersion
+  >,
+  "transport" | "chain"
+> & {
+  signer: TSigner;
+  salt?: bigint;
+  factoryAddress?: Address;
+  initCode?: Hex;
+  initialOwner?: Address;
+  accountAddress?: Address;
+  entryPoint?: EntryPointDef<TEntryPointVersion, Chain>;
+  isGlobalValidation?: boolean;
+  entityId?: bigint;
```

Comment:
I saw a thread from @Blu-J as well around this, but think that PR got closed and comments lost. Where was that thread, there was some interesting discussion in there?

I'm curious what the relationship is between entityId and globalValidation? can you pass in an `entityId != 0` and `isGlobalValidation = true`?

Reply:
Is global validation and entityId always referring to the signer? If so, maybe it makes sense to move some of that logic into a special `Signer` type for MAv2 that wraps an underlying `SmartAccountSigner` type.

Reply:
yeah they're 2 diff vars, but they'll both be used in the nonce. nonceKey has to contain `entityId << 8 | isGlobalValidation`

Reply:
+1 on moving to signer, thought that might make sense. if a dapp collects a sig to add a session key then swaps the signer on the account, they would need to replace the signer + entityId + isGlobalValidation, so I think we can pair it up. Will think a little more if this holds across the 3 MA variants

Reply:
closed those as graphite was doing wonky stuff. comments: 
1. https://github.com/alchemyplatform/aa-sdk/pull/1209#discussion_r1883012331
2. https://github.com/alchemyplatform/aa-sdk/pull/1209#discussion_r1883016912
3. https://github.com/alchemyplatform/aa-sdk/pull/1209#discussion_r1883022802

I especially like the phantom types one to restricting uint32s to get compile time safety instead of runtime safety. I think we can punt on this for the upcoming sprint till Tuesday. I'll take a look + learn more about phantom types after next week when things are less hectic